File: assignments/elixir/atbash-cipher/atbash-cipher_test.exs
Code.load_file("atbash.exs")
ExUnit.start

defmodule AtbashTest do
  use ExUnit.Case, async: true
  doctest Atbash

  test "encode no" do
    assert "ml" == Atbash.encode("no")
  end

  test "encode yes" do
    assert "bvh" == Atbash.encode("yes")
  end

  test "encode OMG" do
    assert "lnt" == Atbash.encode("OMG")
  end

  test "encode O M G" do
    assert "lnt" == Atbash.encode("O M G")
  end

  test "encode long word" do
    assert "nrmwy oldrm tob" == Atbash.encode("mindblowingly")
  end

  test "encode numbers" do
    assert "gvhgr mt123 gvhgr mt" == Atbash.encode("Testing, 1 2 3, testing.")
  end

  test "encode sentence" do
    assert "gifgs rhurx grlm" == Atbash.encode("Truth is fiction.")
  end

  test "encode all the things" do
    plaintext = "The quick brown fox jumps over the lazy dog."
    cipher = "gsvjf rxpyi ldmul cqfnk hlevi gsvoz abwlt"
    assert cipher == Atbash.encode(plaintext)
  end
end

File: assignments/clojure/atbash-cipher/atbash-cipher_test.clj
(ns atbash-cipher.test (:use clojure.test))
(load-file "atbash.clj")

(deftest encode-no
  (is (= "ml" (atbash/encode "no"))))

(deftest encode-yes
  (is (= "bvh" (atbash/encode "yes"))))

(deftest encode-OMG
  (is (= "lnt" (atbash/encode "OMG"))))

(deftest encode-O-M-G
  (is (= "lnt" (atbash/encode "O M G"))))

(deftest encode-long-word
  (is (= "nrmwy oldrm tob" (atbash/encode "mindblowingly"))))

(deftest encode-numbers
  (is (= "gvhgr mt123 gvhgr mt" (atbash/encode "Testing, 1 2 3, testing."))))

(deftest encode-sentence
  (is (= "gifgs rhurx grlm" (atbash/encode "Truth is fiction."))))

(deftest encode-all-the-things
  (let [plaintext "The quick brown fox jumps over the lazy dog."
        cipher "gsvjf rxpyi ldmul cqfnk hlevi gsvoz abwlt"]
    (is (= cipher (atbash/encode plaintext)))))

(run-tests)
